packages/api/src/auth/authHeaders.ts

```diff
@@ -19,11 +13,40 @@ export const getAuthProviderType = (
   return event?.headers[AUTH_PROVIDER_HEADER] as SupportedAuthTypes
 }
 
-type NewClientContext = ClientContext & {
-  user?: object
+export interface AuthorizationHeader {
+  schema: 'Bearer' | 'Basic'
+  token: string
 }
 
-export type AuthToken = null | object | string
+/**
+ * This function returns the Authorization Header Schema and Token
+ * from the client request event headers in AuthorizationHeader
+ * to be decoded or otherwise used in token-based authentication
+ * when required to allow an application to access an API.
+ *
+ * @returns AuthorizationHeader
+ */
+export const getAuthorization = (event: {
+  event: APIGatewayProxyEvent
+}): AuthorizationHeader => {
+  try {
+    const authorizationHeader: AuthorizationHeader = {
+      schema: null,
+      token: null,
+    }
+    const mechanism = event.headers?.authorization?.split(' ')
+    authorizationHeader.schema = mechanism?.[0]
+    authorizationHeader.token = mechanism?.[1]
+
+    if (!mechanism.length === 2 || authorizationHeader.token.length === 0) {
+      throw new Error('Empty authorization token')
+    }
+
+    return authorizationHeader
```

Comment:
Oh, I thought that since:

```
export const getAuthorization = (event: {
  event: APIGatewayProxyEvent
}): AuthorizationHeader
```

had to create that type and return.

Changed to

```
return { schema, token }
```

Reply:
The check needed to be

```
  if (!schema.length || !token.length) {
```

Done.